ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog270.java

```diff
@@ -1282,8 +1282,10 @@ private boolean updateInfraKerberosDescriptor(KerberosDescriptor kerberosDescrip
 
       for (KerberosServiceDescriptor serviceDescriptor : kerberosDescriptor.getServices().values()) {
```

Comment:
It is possible that there are no services as well... you might want to protect again that as well.

Reply:
@rlevas we are calling that above the loop:
```java
services.remove(AMBARI_INFRA_OLD_NAME);
      services.put(AMBARI_INFRA_NEW_NAME, ambariInfraService);
      kerberosDescriptor.setServices(services);
```
based on that we should have a service there

Reply:
awesome... thanks for the clarification.